sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1070,6 +1107,7 @@ public Instant getWatermark() {
 
     @Override
     public CheckpointMark getCheckpointMark() {
+      reportBacklog();
```

Comment:
hmm... all this sounds fairly confusing. Can you point me to metrics doc? A runner is required call getCheckpointMark()?